### DISCUSSION TITLE
请问有什么办法能使config添加deps到一个包

### DISCUSSION THREAD

**Discussion by RimuruChan** [xmake-io/xmake-repo#2564] [Q&A]
https://github.com/xmake-io/xmake-repo/blob/d951fbd89287bd49340d7f39dfacd1b65935bfac/packages/m/mariadb-connector-c/xmake.lua#L26
https://github.com/xmake-io/xmake-repo/blob/d951fbd89287bd49340d7f39dfacd1b65935bfac/packages/m/mariadb-connector-c/xmake.lua#L47
由于这个依赖是后置加载的，导致安装的时候check snippet过不了，必须要保证编译机提前安装过openssl

**Comment by waruqi** [comment]
https://github.com/xmake-io/xmake-repo/blob/d951fbd89287bd49340d7f39dfacd1b65935bfac/packages/m/mariadb-connector-c/xmake.lua#L50 这里依赖已经加了

**Reply by RimuruChan** [reply]
这个是在加载阶段添加的，安装的时候不会生效。如果openssl这个依赖的addrequires在该库后面，就会报错无法链接，无法通过snippets测试

**Reply by waruqi** [reply]
不是，只要你这个 package 里有 add_deps("openssl") 了，on_test 以及用户的 add_packages 阶段集成，都会自动追加这个 openssl 的所有 links 。。跟你外面是否单独 add_requires("openssl") 没有任何关系。。

**Reply by RimuruChan** [reply]
你说得对，但是如果用户没有安装openssl呢？请问如何通过测试？

**Reply by RimuruChan** [reply]
我说的add_requires决定了用户安装依赖的顺序，如果openssl放在了该库的后面，此时用户电脑上并没有openssl库，故无法过编

**Comment by waruqi** [comment]
> 我说的add_requires决定了用户安装依赖的顺序，如果openssl放在了该库的后面，此时用户电脑上并没有openssl库，故无法过编

不是，add_requires 是不决定依赖顺序的，你不管放哪，两个 add_requires 都是平级独立的，仅仅只是告诉用户安装哪些包。。

安装的依赖顺序，是 package 定义的 add_deps 控制的。如果加了  `add_deps("openssl")` ，优先从系统上找。。系统上没有，就会自动拉取源码，编译安装，不会应该系统上没有，就不去安装 link 了。。如果没去装，就提供 `xmake f -cvD` 的完整 logs ，理论上不太可能不装。我这里都可以

**Comment by waruqi** [comment]
也就是说，你仅仅只需要配置 `add_requires("mariadb-connector-c", {configs = {ssl = true}})` ，其他不需要任何配置，外面自己配的 `add_requires("openssl")` 删了，完全用不到。。安装 mariadb-connector-c 时候，会自动安装 openssl，并保证依赖顺序
